src/sys.rs

```diff
@@ -0,0 +1,349 @@
+        }
+    }
+
+    pub fn utc_tm_to_time(rust_tm: &Tm) -> i64 {
+        let mut tm = unsafe { mem::zeroed() };
+        rust_tm_to_tm(rust_tm, &mut tm);
+        unsafe { timegm(&tm) }
+    }
+
+    pub fn local_tm_to_time(rust_tm: &Tm) -> i64 {
+        let mut tm = unsafe { mem::zeroed() };
+        rust_tm_to_tm(rust_tm, &mut tm);
+        unsafe { mktime(&tm) }
+    }
+
+    #[cfg(any(target_os = "macos", target_os = "ios"))]
+    mod mac {
+        use libc::{timeval, timezone, c_int, mach_timebase_info};
+        use std::sync::{Once, ONCE_INIT};
+
+        extern {
+            fn gettimeofday(tp: *mut timeval, tzp: *mut timezone) -> c_int;
+            fn mach_absolute_time() -> u64;
+            fn mach_timebase_info(info: *mut mach_timebase_info) -> c_int;
+        }
+
+        fn info() -> &'static mach_timebase_info {
+            static mut INFO: mach_timebase_info = mach_timebase_info {
+                numer: 0,
+                denom: 0,
+            };
+            static ONCE: Once = ONCE_INIT;
+
+            unsafe {
+                ONCE.call_once(|| {
+                    mach_timebase_info(&mut INFO);
+                });
+                &INFO
+            }
+        }
+
+        pub fn get_time() -> (i64, i32) {
+            use std::ptr;
+            let mut tv = libc::timeval { tv_sec: 0, tv_usec: 0 };
+            unsafe { gettimeofday(&mut tv, ptr::null_mut()); }
+            (tv.tv_sec as i64, tv.tv_usec * 1000)
+        }
+
+        pub fn get_precise_ns() -> u64 {
+            unsafe {
+                let time = mach_absolute_time();
+                let info = info();
+                time * info.numer as u64 / info.denom as u64
+            }
+        }
+    }
+
+    #[cfg(all(not(target_os = "macos"), not(target_os = "ios")))]
+    mod unix {
+        use libc::{self, c_int, timespec};
+
+        #[cfg(all(not(target_os = "android"),
+                  not(target_os = "bitrig"),
+                  not(target_os = "nacl"),
+                  not(target_os = "openbsd")))]
+        #[link(name = "rt")]
+        extern {}
+
+        extern {
+            fn clock_gettime(clk_id: c_int, tp: *mut timespec) -> c_int;
+        }
+
+        pub fn get_time() -> (i64, i32) {
+            let mut tv = libc::timespec { tv_sec: 0, tv_nsec: 0 };
+            unsafe { clock_gettime(libc::CLOCK_REALTIME, &mut tv); }
+            (tv.tv_sec as i64, tv.tv_nsec as i32)
+        }
+
+        pub fn get_precise_ns() -> u64 {
+            let mut ts = libc::timespec { tv_sec: 0, tv_nsec: 0 };
+            unsafe {
+                clock_gettime(libc::CLOCK_MONOTONIC, &mut ts);
+            }
+            (ts.tv_sec as u64) * 1000000000 + (ts.tv_nsec as u64)
+        }
+    }
+}
+
+#[cfg(windows)]
+#[allow(non_snake_case)]
+mod inner {
+    use ::Tm;
+    use std::mem;
+    use std::sync::{Once, ONCE_INIT};
+
+    use winapi::*;
+    use kernel32::*;
+
+    fn frequency() -> libc::LARGE_INTEGER {
```

Comment:
Just `LARGE_INTEGER`. It is provided by `winapi`, no need to refer to `libc` which isn't even used in this module.